lib/membrane_mp4/container/parse_helper.ex

```diff
@@ -54,87 +56,105 @@ defmodule Membrane.MP4.Container.ParseHelper do
     name |> String.trim_trailing(" ") |> String.to_atom()
   end
 
-  defp parse_fields(data, []) do
-    {:ok, {%{}, data}}
+  defp parse_fields(data, [], storage) do
+    {:ok, {%{}, data}, storage}
   end
 
-  defp parse_fields(data, [{name, type} | fields]) do
-    with {:ok, {term, rest}} <- parse_field(data, {name, type}),
-         {:ok, {terms, rest}} <- parse_fields(rest, fields) do
-      {:ok, {Map.put(terms, name, term), rest}}
+  defp parse_fields(data, [{name, type} | fields], storage) do
+    with {:ok, {term, rest}, storage} <- parse_field(data, {name, type}, storage),
+         {:ok, {terms, rest}, storage} <- parse_fields(rest, fields, storage) do
+      {:ok, {Map.put(terms, name, term), rest}, storage}
     end
   end
 
-  defp parse_field(data, {:reserved, reserved}) do
+  defp parse_field(data, {:reserved, reserved}, storage) do
     size = bit_size(reserved)
 
     case data do
-      <<^reserved::bitstring-size(size), rest::bitstring>> -> {:ok, {[], rest}}
+      <<^reserved::bitstring-size(size), rest::bitstring>> -> {:ok, {[], rest}, storage}
       data -> parse_field_error(data, :reserved, expected: reserved)
     end
   end
 
-  defp parse_field(data, {name, subfields}) when is_list(subfields) do
-    case parse_fields(data, subfields) do
-      {:ok, result} -> {:ok, result}
+  defp parse_field(data, {name, {type, store: storage_name}}, storage) do
+    {:ok, result, storage} = parse_field(data, {name, type}, storage)
+    {value, _rest} = result
+    storage = Map.put(storage, storage_name, value)
+    {:ok, result, storage}
+  end
+
+  defp parse_field(data, {name, {type, when: condition}}, storage) do
```

Comment:
This approach doesn't work when you have `store` and `when` at the same time. It may get more important as we add more directives.

Reply:
indeed, done